packages/client-proxy/main.go

```diff
@@ -125,15 +135,15 @@ func main() {
 			// this probably shouldn't happen...
 			logger.Error("http service exited without error", zap.Int("port", port))
 		}
-		wg.Done()
 	}()
 
 	// Proxy request to the correct node
 	server := http.Server{Addr: fmt.Sprintf(":%d", port)}
```

Comment:
I think we should declare the `http.Server` as pointers (e.g. "&http.Server{}`). I think it works ok  in it's current form, but if we accidentally pass it without it being a pointer, then it ends up being a shallow copy, and that seems bad.